src/rebalancer/bridges/deprecated/LZRsEthBridge.sol

```diff
@@ -0,0 +1,209 @@
+    error LZBridge_NoRemote();
+
+    constructor(address _roles, address _endpoint) BaseBridge(_roles) {
+        require(_endpoint != address(0), "endpoint=0");
+        endpoint = _endpoint;
+    }
+
+    // ----------- ADMIN ------------
+    function setBridgeContract(address underlying, address bridgeContract) external onlyBridgeConfigurator {
+        bridgeContracts[underlying] = bridgeContract;
+        emit BridgeContractSet(underlying, bridgeContract);
+    }
+
+    // ----------- VIEW ------------
+    function getFee(uint32 _dstChainId, bytes memory _message, bytes memory)
+        external
+        view
+        returns (uint256)
+    {
+        require(_dstChainId > 0, LZBridge_ChainNotRegistered());
+        (MessagingFee memory fees,) = _quote(_dstChainId, _message);
+        return fees.nativeFee;
+    }
+
+    // ----------- EXTERNAL ------------
+    function sendMsg(
+        uint256 _extractedAmount,
+        address _market,
+        uint32 _dstChainId,
+        address _token,
+        bytes memory _message,
+        bytes memory
+    )
+        external
+        payable
+        onlyRebalancer
+    {
+        require(_dstChainId > 0, LZBridge_ChainNotRegistered());
+
+        (address market,,,) = abi.decode(_message, (address, uint256, uint256, bytes));
+        require(_market == market, LZBridge_DestinationMismatch());
+
+        (MessagingFee memory fees, SendParam memory sendParam) = _quote(_dstChainId, _message);
+        if (msg.value < fees.nativeFee) revert LZBridge_NotEnoughFees();
+        require(_extractedAmount == sendParam.amountLD, BaseBridge_AmountMismatch());
+
+        address underlying = ImTokenMinimal(_market).underlying();
+        if (underlying != _token) revert LZBridge_TokenMismatch();
+
+        address bridgeContract = bridgeContracts[underlying];
+        if (bridgeContract == address(0)) bridgeContract = underlying;
+
+        // pull from rebalancer
+        IERC20(_token).safeTransferFrom(msg.sender, address(this), sendParam.amountLD);
+
+        // adapter path: deposit legacy -> mint OFT to this contract (on source)
+        if (bridgeContract != underlying) {
+            try ILayerZeroOFTWrapper(underlying).allowedTokens(bridgeContract) returns (bool ok) {
+                require(ok, LZBridge_DifferentInnerToken());
+            } catch {
+                revert LZBridge_NoOft();
+            }
+            SafeApprove.safeApprove(underlying, bridgeContract, sendParam.amountLD);
+            ILayerZeroOFTWrapper(underlying).withdraw(bridgeContract, sendParam.amountLD);
+
+            if (IERC20(bridgeContract).balanceOf(address(this)) < sendParam.amountLD) {
+                revert LZBridge_AmountMismatch();
+            }
+        }
+
+        // send OFT to destination bridge; compose will withdraw -> legacy -> market
+        (MessagingReceipt memory msgReceipt,) =
+            ILayerZeroOFT(bridgeContract).send{value: fees.nativeFee}(sendParam, fees, msg.sender);
+
```

Comment:
`sendMsg` doesn’t refund excess `msg.value`: it only forwards `fees.nativeFee` and keeps any overpayment on the bridge contract. Callers who overpay for safety will strand native ETH. 

Recommend refunding `msg.value - fees.nativeFee` to the payer or requiring exact payment.